commands/diskstat.rb

```diff
@@ -0,0 +1,86 @@
+require 'io/console'
+require_relative '../lib/const'
+require_relative '../lib/color'
+require_relative '../lib/convenience_functions'
+require_relative '../lib/misc_functions'
+
+class Command
+  def self.diskstat(show_all)
+    color_list = [
+      %i[lightred no_bold],
+      %i[lightcyan no_bold],
+      %i[lightpurple no_bold],
+      %i[yellow no_bold],
+      %i[lightblue no_bold],
+      %i[lightgreen no_bold],
+      :red,
+      :cyan,
+      :purple,
+      :orange,
+      :blue,
+      :green
+    ]
+
+    terminal_w = !IO.console&.console_mode || IO.console&.winsize == [0, 0] ? 80 : IO.console&.winsize&.last
+
+    # Calculate disk size for all packages and sort in ascending order
+    size_of_all_packages = Dir["#{CREW_META_PATH}/*.filelist"].map do |filelist|
+      pkg_name = File.basename(filelist, '.filelist')
+      size     = ConvenienceFunctions.read_filelist(filelist, always_calcuate_from_disk: true).first
+
+      [pkg_name, size]
+    end.sort_by(&->((_, size)) { -size })
+
+    total_size     = size_of_all_packages.sum(&:last).to_f # Total size of all installed packages
+    other_size     = total_size                            # Total size of all installed packages, excluding the top 12 one
+    bar_length     = 0
+    bar_components = size_of_all_packages[..11].map.with_index do |(_, size), i|
```

Comment:
What are you thoughts about making the 11 configurable with a switch option and/or environment variable?

Reply:
All packages will be printed below, but only the top 12 are inside the bar chart.

EDIT: Just found a way to display up to 24 different colors